code/renderers/vue3/src/docs/sourceDecorator.ts

```diff
@@ -36,89 +43,80 @@ const skipSourceRender = (context: StoryContext<Renderer>) => {
-    children: component?.children || null,
-    attributes: component?.attributes || component?.attrs || null,
-  };
-}
 /**
  *
  * @param _args
  * @param argTypes
  * @param byRef
  */
-function generateAttributesSource(_args: Args, argTypes: ArgTypes, byRef?: boolean): string {
-  // create a copy of the args object to avoid modifying the original
-  const args = { ..._args };
-  // filter out keys that are children or slots, and convert event keys to the proper format
-  const argsKeys = Object.keys(args)
-    .filter(
-      (key: any) =>
-        ['children', 'slots'].indexOf(argTypes[key]?.table?.category) === -1 || !argTypes[key] // remove slots and children
-    )
-    .map((key) => {
-      const akey =
-        argTypes[key]?.table?.category !== 'events' // is event
-          ? key
-              .replace(/([A-Z])/g, '-$1')
-              .replace(/^on-/, 'v-on:')
-              .replace(/^:/, '')
-              .toLowerCase()
-          : `v-on:${key}`;
-      args[akey] = args[key];
-      return akey;
+export function generateAttributesSource(
+  tempArgs: (AttributeNode | DirectiveNode)[],
+  args: Args,
+  argTypes: ArgTypes,
+  byRef?: boolean
+): string {
+  return Object.keys(tempArgs)
+    .map((key: any) => {
+      return evalExp(tempArgs[key].loc.source.replace(/\$props/g, 'args'), omitEvent(args));
     })
-    .filter((key, index, self) => self.indexOf(key) === index); // remove duplicated keys
-
-  const camelCase = (str: string) => str.replace(/-([a-z])/g, (g) => g[1].toUpperCase());
-  const source = argsKeys
-    .map((key) =>
-      generateAttributeSource(
-        byRef && !key.includes(':') ? `:${key}` : key,
-        byRef && !key.includes(':') ? camelCase(key) : args[key],
-        argTypes[key]
-      )
-    )
     .join(' ');
-
-  return source;
 }
+/**
+ * map attributes and directives
+ * @param props
+ */
+function mapAttributesAndDirectives(props: Args) {
+  const tranformKey = (key: string) => (key.startsWith('on') ? key : kebabCase(key));
+  return Object.keys(props).map(
+    (key) =>
+      ({
+        name: 'bind',
+        type: ['v-', '@', 'v-on'].includes(key) ? 7 : 6, // 6 is attribute, 7 is directive
+        arg: { content: tranformKey(key), loc: { source: tranformKey(key) } }, // attribute name or directive name (v-bind, v-on, v-model)
+        loc: { source: attributeSource(tranformKey(key), props[key]) }, // attribute value or directive value
+        exp: { isStatic: false, loc: { source: props[key] } }, // directive expression
+        modifiers: [''],
+      } as unknown as AttributeNode)
+  );
+}
+/**
+ *  map slots
+ * @param slotsProps
+ */
+function mapSlots(slotsProps: Args, generateComponentSource: (v: VNode) => string): TextNode[] {
+  return Object.keys(slotsProps).map((key) => {
+    const slot = slotsProps[key];
+    let slotContent = '';
+
+    if (typeof slot === 'function') {
+      const slotVNode = slot();
```

Comment:
This line crashes when using scoped slots without render function:

<img width="1695" alt="image" src="https://user-images.githubusercontent.com/1035299/225333243-bb9717c0-a053-4a02-8327-af1374bbe3cf.png">

```
TypeError: Cannot destructure property 'text' of 'undefined' as it is undefined.
    at default (ScopedSlots.stories.js:15:7)
    at config.mjs:7671:25
    at Array.map (<anonymous>)
    at mapSlots (config.mjs:7667:34)
    at generateComponentSource (config.mjs:7745:81)
    at config.mjs:7753:61
    at Array.map (<anonymous>)
    at generateTemplateSource (config.mjs:7753:36)
    at generateSource (config.mjs:7777:29)
    at sourceDecorator (config.mjs:7759:3)
```

If I put a default value, it doesn't crash:

<img width="1689" alt="image" src="https://user-images.githubusercontent.com/1035299/225333623-e6434c26-1e10-45c3-9a33-8ae7cd6d393c.png">

Reply:
@chakAs3 I updated the test a bit. It work now in canvas mode, but still break in docs mode:

<img width="1070" alt="image" src="https://user-images.githubusercontent.com/1035299/225616707-6c02de40-c8f5-4641-898c-9d5a57459aa6.png">

Reply:
It seems that only templates are tested there, but not if you use the default render function.